qiita_db/test/test_user.py

```diff
@@ -353,6 +361,53 @@ def test_change_forgot_password_bad_code(self):
         self.assertEqual(obsbool, False)
         self._check_pass("password")
 
+    def test_messages(self):
+        user = User('test@foo.bar')
+        obs = user.messages()
+        exp_msg = [
+            (1, 'message 1'),
+            (2, 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. '
+                'Pellentesque sed auctor ex, non placerat sapien. Vestibulum '
+                'vestibulum massa ut sapien condimentum, cursus consequat diam'
+                ' sodales. Nulla aliquam arcu ut massa auctor, et vehicula '
+                'mauris tempor. In lacinia viverra ante quis pellentesque. '
+                'Nunc vel mi accumsan, porttitor eros ut, pharetra elit. Nulla'
+                ' ac nisi quis dui egestas malesuada vitae ut mauris. Morbi '
+                'blandit non nisl a finibus. In erat velit, congue at ipsum '
+                'sit amet, venenatis bibendum sem. Curabitur vel odio sed est '
+                'rutrum rutrum. Quisque efficitur ut purus in ultrices. '
+                'Pellentesque eu auctor justo.'),
+            (3, 'message <a href="#">3</a>')]
+        self.assertEqual([(x[0], x[1]) for x in obs], exp_msg)
+        self.assertTrue(all(x[2] < datetime.now() for x in obs))
+        self.assertFalse(all(x[3] for x in obs))
+
+        obs = user.messages(1)
+        exp_msg = ['message 1']
+        self.assertEqual([x[1] for x in obs], exp_msg)
+
+    def test_mark_messages(self):
+        user = User('test@foo.bar')
+        user.mark_messages([1, 2])
+        obs = user.messages()
+        exp = [True, True, False]
+        self.assertEqual([x[3] for x in obs], exp)
+
+        user.mark_messages([1], read=False)
+        obs = user.messages()
+        exp = [False, True, False]
+        self.assertEqual([x[3] for x in obs], exp)
+
+    def test_delete_messages(self):
```

Comment:
Can you add a test for making sure that admin messages cannot be deleted by the user?